Whiskey/Tasks/New-WhiskeyGitHubRelease.ps1

```diff
@@ -0,0 +1,186 @@
+        $TaskParameter
+    )
+
+    Set-StrictMode -Version 'Latest'
+    Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
+
+    $apiKeyID = $TaskParameter['ApiKeyID']
+    if( -not $apiKeyID )
+    {
+        Stop-WhiskeyTask -TaskContext $TaskContext -Message ('Property "ApiKeyID" is mandatory. It should be set to the ID of the API key to use when talking to the GitHub API. API keys are added to your build with the "Add-WhiskeyApiKey" function.')
+        return
+    }
+
+    $apiKey = Get-WhiskeyApiKey -Context $TaskContext -ID $apiKeyID -PropertyName 'ApiKeyID'
+    $headers = @{
+	                Authorization = 'Basic ' + [Convert]::ToBase64String([Text.Encoding]::UTF8.GetBytes($apiKey + ":x-oauth-basic"))
+                }
+    $repositoryName = $TaskParameter['RepositoryName']
+    if( -not $repositoryName )
+    {
+        Stop-WhiskeyTask -TaskContext $TaskContext -Message ('Property "RepositoryName" is mandatory. It should be the owner and repository name of the repository you want to access as a URI path, e.g. OWNER/REPO.')
+        return
+    }
+
+    if( $repositoryName -notmatch '^[^/]+/[^/]+$' )
+    {
+        Stop-WhiskeyTask -TaskContext $TaskContext -Message ('Property "RepositoryName" is invalid. It should be the owner and repository name of the repository you want to access as a URI path, e.g. OWNER/REPO.')
+        return
+    }
+
+    $baseUri = [uri]'https://api.github.com/repos/{0}' -f $repositoryName
+
+    [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor [System.Net.SecurityProtocolType]::Tls12
+
+    function Invoke-GitHubApi
+    {
+        [CmdletBinding(DefaultParameterSetName='NoBody')]
+        param(
+            [Parameter(Mandatory=$true)]
+            [uri]
+            $Uri,
+
+            [Parameter(Mandatory=$true,ParameterSetName='FileUpload')]
+            [string]
+            $ContentType,
+
+            [Parameter(Mandatory=$true,ParameterSetName='FileUpload')]
+            [string]
+            $InFile,
+
+            [Parameter(Mandatory=$true,ParameterSetName='JsonRequest')]
+            $Parameter,
+
+            [Microsoft.PowerShell.Commands.WebRequestMethod]
+            $Method = [Microsoft.PowerShell.Commands.WebRequestMethod]::Post
+        )
+
+        $optionalParams = @{ }
+        if( $PSCmdlet.ParameterSetName -eq 'JsonRequest' )
+        {
+            if( $Parameter )
+            {
+                $optionalParams['Body'] = $Parameter | ConvertTo-Json
+    	        Write-WhiskeyVerbose -Context $TaskContext -Message $optionalParams['Body']
+            }
+            $ContentType = 'application/json'
+        }
+        elseif( $PSCmdlet.ParameterSetName -eq 'FileUpload' )
+        {
+            $optionalParams['InFile'] = $InFile
+        }
+
+        try
+        {
+	        Invoke-RestMethod -Uri $Uri -Method $Method -Headers $headers -ContentType $ContentType @optionalParams

```

Comment:
This was intentional so `$headers` was defined only once and the encoded API key is only in one place in memory instead of a bunch of abandoned, un-garbage collected strings.